rust/processor/src/processors/fungible_asset_processor.rs

```diff
@@ -453,294 +454,318 @@ async fn parse_v2_coin(
-                            tracing::error!(
-                            transaction_version = txn_version,
-                            index = index,
-                                error = ?e,
-                            "[Parser] error parsing fungible metadata v2");
-                            panic!("[Parser] error parsing fungible metadata v2");
-                        })
-                    {
-                        fungible_asset_metadata.insert(fa_metadata.asset_type.clone(), fa_metadata);
-                    }
-                    if let Some((balance, curr_balance)) =
-                        FungibleAssetBalance::get_v2_from_write_resource(
-                            write_resource,
-                            index as i64,
-                            txn_version,
-                            txn_timestamp,
-                            &fungible_asset_object_helper,
-                        )
-                        .await
-                        .unwrap_or_else(|e| {
-                            tracing::error!(
-                            transaction_version = txn_version,
-                            index = index,
-                                error = ?e,
-                            "[Parser] error parsing fungible balance v2");
-                            panic!("[Parser] error parsing fungible balance v2");
-                        })
-                    {
-                        fungible_asset_balances.push(balance);
-                        current_fungible_asset_balances
-                            .insert(curr_balance.storage_id.clone(), curr_balance);
-                    }
-                },
-                Change::WriteTableItem(table_item) => {
-                    if let Some(coin_supply) = CoinSupply::from_write_table_item(
-                        table_item,
-                        txn_version,
-                        txn_timestamp,
-                        txn_epoch,
-                    )
-                    .unwrap()
-                    {
-                        all_coin_supply.push(coin_supply);
-                    }
-                },
-                _ => {},
+                        .unwrap()
+                        {
+                            all_coin_supply.push(coin_supply);
+                        }
+                    },
+                    _ => {},
+                }
             }
-        }
+            (
+                fungible_asset_activities,
+                fungible_asset_balances,
+                all_coin_supply,
+                current_fungible_asset_balances,
+                fungible_asset_metadata,
+            )
+        })
+        .collect();
+
+    for (faa, fab, acs, cfab, fam) in data {
+        fungible_asset_activities.extend(faa);
+        fungible_asset_balances.extend(fab);
+        all_coin_supply.extend(acs);
+        current_fungible_asset_balances.extend(cfab);
+        fungible_asset_metadata.extend(fam);
```

Comment:
Is this necessary?

Reply:
I don't want to change the insert_to_db interface for now.